eth/api_backend.go

```diff
@@ -74,7 +74,7 @@ func (b *EthApiBackend) StateAndHeaderByNumber(blockNr rpc.BlockNumber) (ethapi.
 	// Pending state is only known by the miner
 	if blockNr == rpc.PendingBlockNumber {
 		block, state := b.eth.miner.Pending()
-		return EthApiState{state}, block.Header(), nil
+		return EthApiState{state.Copy()}, block.Header(), nil
```

Comment:
It makes only a copy when the node is [mining](https://github.com/ethereum/go-ethereum/blob/develop/miner/worker.go#L162). Otherwise it doesn't. I'm not sure why this distinction was made in the past.